### ISSUE TITLE
[BUG] Support for mixed precision training with TF

### ISSUE THREAD

**Issue by gabrielspmoreira** [NVIDIA-Merlin/models#215]
### Bug description
When enabling mixed precision training with TF (for higher performance using GPU's TensorCodes), the train loss eventually get's unstable and results in nan loss.

#### Example
```python
tf.keras.mixed_precision.set_global_policy("mixed_float16")
```
